File: python/24. Swap Nodes in Pairs/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:

        if not head or not head.next:
            return head

        current = head
        next = current.next
        temp = next.next
        next.next = current
        current.next = self.swapPairs(temp)

        return next

File: java/24. Swap Nodes in Pairs/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {

        if (head == null || head.next == null) return head;

        ListNode current = head;
        ListNode next = head.next;


        ListNode temp = next.next;
        next.next = current;
        current.next = swapPairs(temp);

        return next;
    }
}